chef/cookbooks/barclamp/libraries/barclamp_library.rb

```diff
@@ -264,6 +264,23 @@ def self.unclaimed(node, include_mounted=false)
           end
         end
 
+        # return the disks that are unclaimed and prepared for given role (or with no role)
+        # (sort the result so the ones with exact role come first)
```

Comment:
I can also let this function return 2 lists, one containing disks with selected role, and one with no roles.
Than it's up to users of this function, how to handle it.